Unit/functional test improvements to enable CTSM SystemTests

samsrabin opened the pull request:
- Gets actual path to default config file to enable running from anywhere.
- Adds an error check that would have helped in my troubleshooting.
- Adds optional `--config-file` argument.
- Adds netCDF file required for unit testing.

CTSM PR here: ESCOMP/CTSM#3251

samsrabin committed:
run_unit_testing.py: Enable running from anywhere.

samsrabin committed:
run_unit_tests.py: Fail if config_file doesn't exist or is a dir.

samsrabin committed:
run_unit_tests.py: Add optional --config-file argument.

samsrabin commented:
@glemieux I've added a netCDF file required by the functional unit testing, via git LFS, which seems new for the FATES repo. I wanted to check in with you on that—does it seem acceptable? This means that any _tracked_ netCDF files will be handled by git LFS, but I left `*.nc` in the .gitignore file, so actually tracking a netCDF file will require someone to use the `--force` flag with `git add`.

glemieux commented:
> @glemieux I've added a netCDF file required by the functional unit testing, via git LFS, which seems new for the FATES repo. I wanted to check in with you on that—does it seem acceptable? This means that any _tracked_ netCDF files will be handled by git LFS, but I left `*.nc` in the .gitignore file, so actually tracking a netCDF file will require someone to use the `--force` flag with `git add`.

I think that makes sense.  I like keeping the `.nc` file in the `.gitignore` but allowing a way to force it for testing.  Thanks for checking.

samsrabin commented:
Thanks! Converting to draft while I work on a SystemTest for `run_functional_tests.py`.

samsrabin commented:
Hmm. git lfs is giving me trouble. Reverted and testing some more.

samsrabin committed:
Functional test fix: DATM file now found when using -r.

samsrabin committed:
Functional tests now work if called from somewhere other than testing/.

samsrabin committed:
Functional tests: Simplify an error.

samsrabin committed:
run_functional_tests.py: Add optional --config-file argument.

samsrabin committed:
Functional tests: Error if Fortran executable fails.

samsrabin committed:
Functional tests: Fortran now errors instead of just stopping.

samsrabin committed:
Functional tests: Download datm_file if needed. KLUDGE!

This is a kludge because it (a) downloads the datm_file from my Dropbox and (b) requires an extra line in each config file section. Revert this change once BONA_datm.nc is on Git LFS and that's working consistently.

samsrabin commented:
Closing to replace with NGEET/fates#1426.
